packages/contracts-bedrock/scripts/checks/check-interfaces.sh

```diff
@@ -0,0 +1,106 @@
+# It compares the ABIs of interfaces (files starting with 'I') with their implementation contracts,
+# excluding constructors and certain predefined files. The script reports any differences found
+# and exits with an error if inconsistencies are detected.
+
+# Grab the directory of the contracts-bedrock package
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+CONTRACTS_BASE=$(dirname "$(dirname "$SCRIPT_DIR")")
+
+# Define the files to exclude (patterns can be used)
+EXCLUDE_FILES=(
+  # External dependencies
+  "IMulticall3"
+  "IERC20"
+  "IERC721"
+  "IERC721Enumerable"
+  "IERC721Metadata"
+  "IERC721Upgradeable"
+  "IERC721Receiver"
+  "IERC1271"
+  "IERC165"
+  "IVotes"
+  "IBeacon"
+  "IProxyCreationCallback"
+  "IEAS"
+  "ISchemaResolver"
+  "ISchemaRegistry"
+  "IAutomate"
+  "IGelato1Balance"
+
+  # TODO: Interfaces that need to be fixed
+  "IDisputeGameFactory"
+  "IPreimageOracle"
+  "IOptimismMintableERC721"
+  "IFaultDisputeGame"
+  "IInitializable"
+  "IAddressManager"
+  "IWETH"
+  "IDelayedWETH"
+  "IAnchorStateRegistry"
+  "ICrossL2Inbox"
+  "IL2ToL2CrossDomainMessenger"
+)
+
+# Find all JSON files in the forge-artifacts folder
+JSON_FILES=$(find "$CONTRACTS_BASE/forge-artifacts" -type f -name "*.json")
+
+# Initialize a flag to track if any differences are detected
+differences_detected=false
+
+# Iterate over all JSON files
+for interface_file in $JSON_FILES; do
+    # Extract contract kind and name
+    contract_kind=$(jq -r '.ast.nodes[] | select(.nodeType == "ContractDefinition") | .contractKind' < "$interface_file")
+    contract_name=$(jq -r '.ast.nodes[] | select(.nodeType == "ContractDefinition") | .name' < "$interface_file")
+
+    # Skip excluded files
+    skip=false
+    for exclude in "${EXCLUDE_FILES[@]}"; do
+        if [[ "$interface_file" == *"$exclude"* ]]; then
+            skip=true
+            break
+        fi
+    done
+
+    # Continue to the next file if it should be skipped
+    if [ "$skip" = true ]; then
+        continue
+    fi
+
+    # If contract kind is "interface" and name starts with "I"
+    if [[ "$contract_kind" == "interface" && "$contract_name" == I* ]]; then
+        # Construct the corresponding contract name by removing the leading "I"
+        contract_basename=${contract_name:1}
+        corresponding_contract_file="$CONTRACTS_BASE/forge-artifacts/$contract_basename.sol/$contract_basename.json"
+
+        # Check if the corresponding contract file exists
+        if [ -f "$corresponding_contract_file" ]; then
+            # Extract and compare ABIs excluding constructors
+            interface_abi=$(jq '[.abi[] | select(.type != "constructor")]' < "$interface_file")
+            contract_abi=$(jq '[.abi[] | select(.type != "constructor")]' < "$corresponding_contract_file")
+
+            # Compare the filtered ABIs using jd
+            diff_output=$(jd --set <(echo "$interface_abi") <(echo "$contract_abi"))
```

Comment:
I wonder if there is a way to use `jq` for this

Also link to `jd` docs for reviewers: https://pkg.go.dev/github.com/josephburnett/jd

Reply:
Modified to use `jq` instead.